src/Schema/Comparator.php

```diff
@@ -199,8 +212,18 @@ public function diffSequence(Sequence $sequence1, Sequence $sequence2)
      *
      * @throws SchemaException
      */
-    public function diffTable(Table $fromTable, Table $toTable)
+    public function diffTable(Table $fromTable, Table $toTable, ?AbstractPlatform $platform = null)
```

Comment:
I believe we could use this as an opportunity to introduce better method names. `diffTable()` could be renamed to `compareTables()`. This way, we won't have to introduce an optional parameter but could introduce the whole old method.

Reply:
Yesterday I openned https://github.com/doctrine/dbal/pull/4707 to introduce a more consistent Comparator API.

But as @ausi commented, it will make merging this PR harder, mainly because of the optional `$platform` parameter.

I think injecting the platform as a constructor parameter when creating a Comparator object would help with avoiding conflcts and would make sense design wise (comparing depends on the platform).

Actually, it is precisely for enabling this type of API evolution that https://github.com/doctrine/dbal/pull/4707 deprecates static calls to `compareSchemas()`, as stated in https://github.com/doctrine/dbal/pull/4704#discussion_r667042163.

What do you think ?

Reply:
> But as @ausi commented, it will make merging this PR harder, mainly because of the optional `$platform` parameter on all public methods.

Let's not worry about that. I can help resolve all the conflicts once they occur.

> I think injecting the platform as a constructor parameter when creating a Comparator object would help with avoiding conflcts and would make sense design wise: it seems comparing schemas depends on the platform, given all the linked pull requests.

Whether it's injected in the constructor or passed as an argument, it's the same design-wise: the comparator does depend on the platform. The question is, whether or not a given comparator instance needs be usable with different platforms, and the answer is "probably not". I think we can move the platform dependency a constructor argument as originally proposed.

Reply:
The `compareColumns()` method now returns `bool` in this pull request. This would conflict with #4707 (but only if there would actually be a version release between the two pull requests).